src/Aspire.Hosting/VersionChecking/VersionCheckService.cs

```diff
@@ -0,0 +1,178 @@
+        }
+
+        if (latestVersion == null || IsVersionGreaterOrEqual(_appHostVersion, latestVersion))
+        {
+            // App host version is greater than the latest version so exit.
+            return;
+        }
+
+        if (TryGetConfigVersion(IgnoreVersionKey, out var ignoreVersion))
+        {
+            if (IsVersionGreaterOrEqual(ignoreVersion, latestVersion))
+            {
+                // Ignored version is greater or equal to latest version so exit.
+                _logger.LogDebug("Ignoring version {Version} as it is less than or equal to the ignored version {IgnoreVersion}.", latestVersion, ignoreVersion);
+                return;
+            }
+        }
+
+        if (IsVersionGreater(latestVersion, storedKnownLatestVersion) || storedKnownLatestVersion == null)
+        {
+            // Latest version is greater than the stored known latest version, so update it.
+            SecretsStore.TrySetUserSecret(_options.Assembly, KnownLastestVersionDateKey, latestVersion.ToString());
+        }
+
+        var result = await _interactionService.PromptMessageBarAsync(
+            title: "Update now",
+            message: $"Aspire {latestVersion} is available.",
+            options: new MessageBarInteractionOptions
+            {
+                LinkText = "Upgrade instructions",
+                LinkUrl = "https://aka.ms/dotnet/aspire/update-latest",
+                PrimaryButtonText = "Ignore"
+            },
+            cancellationToken: stoppingToken).ConfigureAwait(false);
+
+        // True when the user clicked the primary button (Ignore).
+        if (result.Data)
+        {
+            _logger.LogDebug("User chose to ignore version {Version}.", latestVersion);
+            SecretsStore.TrySetUserSecret(_options.Assembly, IgnoreVersionKey, latestVersion.ToString());
+        }
+
+        return;
+    }
+
+    public static bool IsVersionGreaterOrEqual(SemVersion? version1, SemVersion? version2)
+    {
+        if (version1 == null || version2 == null)
+        {
+            return false;
+        }
+        return SemVersion.ComparePrecedence(version1, version2) >= 0;
+    }
+
+    public static bool IsVersionGreater(SemVersion? version1, SemVersion? version2)
+    {
+        if (version1 == null || version2 == null)
+        {
+            return false;
+        }
+        return SemVersion.ComparePrecedence(version1, version2) > 0;
+    }
+
+    private bool TryGetConfigVersion(string key, [NotNullWhen(true)] out SemVersion? knownLatestVersion)
+    {
+        if (_configuration[key] is string latestVersionString &&
+            SemVersion.TryParse(latestVersionString, out knownLatestVersion))
+        {
+            return true;
+        }
+
+        knownLatestVersion = null;
+        return false;
+    }
+}
+
+#pragma warning restore ASPIREINTERACTION001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
```

Comment:
```suggestion
```

What's the value in restoring at the end of the file?

Reply:
The sense of satisfaction of a job well done.